main.cpp

```diff
@@ -1,25 +1,57 @@
 #include <iostream>
+#include <getopt.h>
 #include "common.h"
-#include "md5-duplicate-detection.h"
 #include "encrypt-decrypt.h"
 #include "files.h"
 
-int main() {
-    std::cout << "Hello, World!" << std::endl;
-    /*
-    std::string directory = "some directory";
+int main(int argc, char *argv[]) {
+
+    int opt;
+    const char *directory;
     std::vector<fs::path> files;
-    getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
 
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
+    while (true) { //getopt creates a data-structure that can be looped through from the argv array to process the parameters and their arguments
+
+        int option_index = 0;
+        static struct option long_options[] =
+                {
+                        {"help",     no_argument, NULL,  'h' },
+                        {"encrypt",  no_argument, NULL,  'e' },
+                        {"decrypt",  no_argument, NULL,  'd' },
+                        {NULL,      0,    NULL,   0  }
+                };
```

Comment:
I am not sure what a static struct is, but it can be moved outside of the while loop or main even since it is a struct.

Reply:
moved the struct and it doesn't need to be static so I changed that and declared it outside the loop which made more sense than in the loop